hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -214,6 +214,34 @@
       tests: ["chip_sw_spi_device_tpm"]
     }
 
+    {
+      name: chip_sw_spi_device_output_when_disabled_or_sleeping
+      desc: '''Verify spi_device output values when spi_device is disabled or the chip is sleeping.
+
+               SW needs to be able to set the SPI output value when spi_device is disabled or the
+               chip is sleeping, to either all-zeros or all-ones, depending on integration
+               requirements.  The following scenarios have to be verified:
+
+               After power-on reset:
+               - SW to configure pinmux retention logic so that the chip pins connected to
+                 spi_device outputs are (a) always zero or (b) always one (SW needs to be able to
+                 choose between a and b).
+               - DV environment to check that SPI outputs match configuration by SW.
+
+               Going to sleep:
+               - SW to disable spi_device, wait until CSb is high, configure pinmux retention logic
+                 as it would after POR, and put chip to sleep.
+               - DV environment to check that SPI outputs match configuration by SW.
+
+               Wake up from sleep:
+               - DV environment to wake chip from sleep.
+               - SW to enable spi_device and disable retention logic.
+               - DV environment to check that SPI transactions work as usual.
+             '''
+      stage: V3
+      tests: []
```

Comment:
If we agree on adding this testpoint, I'll create an issue and link it in a TODO comment after the empty `tests` list.

Reply:
If pinmux retention logic is used (as now described in the testplan), the outputs are actively driven to zero or one. I see your point that this could create drive conflicts. The drive strength of the pad can be configured to the lowest value, but that may not be sufficient to resolve the drive conflict, right? This may be especially critical for wake ups, where, e.g., the wakeup detector waits for a falling edge caused by the host driving the wire to zero but cannot observe it because retention logic simultaneously drives the wire to one.

To solve this, can we instead configure pinmux retention logic to let the pad float (high-Z) and activate the pad-internal pull-up or pull-down mechanism? Or will this not work either, because the internal pull up would compete with the PCB pull up?